src/libostree/ostree-repo-libarchive.c

```diff
@@ -480,6 +480,12 @@ file_to_archive_entry_common (GFile         *root,
   g_autoptr(GVariant) xattrs = NULL;
   time_t ts = (time_t) opts->timestamp_secs;
 
+  if (opts->path_prefix && opts->path_prefix[0])
+    {
+      g_autofree char *old_pathstr = pathstr;
+      pathstr = g_build_filename (opts->path_prefix, old_pathstr, NULL);
```

Comment:
One thing that is different from `git archive --prefix`  is that we're automatically inserting e.g. a `/` if the host platform is Unix.  In the unlikely event of a Windows port, we'll insert `\`.  Which in turn would make `--prefix` unusable on Windows to generate Unix archives.

So it's more annoying, but strictly more powerful, to use `g_strconcat()` instead.  The argument that it matches git is important for OSTree.